### PR TITLE
Add "Report Harmful Language" form

### LINKED ISSUE #3331: Create the Report Harmful Language Form - email to new harmful content
### User story
part of #3272 

### Concrete example
proposed form: https://docs.google.com/document/d/1HPOUhTo6T9Zi57G0z-hsKIJCKU1u3k_KZgyJyuHunJw/edit

### Implementation notes, if any
- Send email to the new harmful content queue
- Can base changes on ["Ask a Question" form code](https://github.com/pulibrary/orangelight/commit/34a97f827b5321b55182676010be5a4d328db0e8)

### PR DESCRIPTION
**Pull request by maxkadel**
Closes #3331

### CODE REVIEW

**Comment by kevinreiss** [comment]
Looking at the form on staging it's working well. The only UX concern I'd have is the amount of text displayed before the form. I wonder if the committee might consider shortening it.

**Comment by christinach** [comment]
The group IRMWG will review the form. This should not block the PR.

File: app/views/feedback/report_harmful_language.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<%= render Blacklight::System::ModalComponent.new do |component| %>
```
Comment (sandbergja) [inline app/views/feedback/report_harmful_language.html.erb:1]: Nice!

File: spec/forms/report_harmful_language_form_spec.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      form = described_class.new(minimal_valid_attributes)
+
+      form.submit
+      expect(ActionMailer::Base.deliveries.length).to eq 1
+      expect(form).to be_submitted
+
+      mail = ActionMailer::Base.deliveries.first
+      expect(mail.subject).to eq "[Possible Harmful Language] Example Record"
+      expect(mail.from).to eq ["test-harmful-content@princeton.edu"]
+      # The behavior for empty Name and Email are TBD at the moment
```
Comment (sandbergja) [inline spec/forms/report_harmful_language_form_spec.rb:80]: Can these be un-commented now?

File: spec/mailers/contact_mailer_spec.rb [hunk 3]
```diff
@@ -34,4 +34,66 @@
...
+
+      it "renders the headers" do
+        expect(mail.subject).to eq("[Possible Harmful Language] Example Record")
+        expect(mail.to).to eq(["test-harmful-content@princeton.edu"])
+        expect(mail.from).to eq(["test-harmful-content@princeton.edu"])
+      end
+
+      it "renders the body" do
+        # The behavior for empty Name and Email are TBD at the moment
+        # expect(mail.body.encoded).to have_content('Name: Test')
```
Comment (sandbergja) [inline spec/mailers/contact_mailer_spec.rb:91]: Ditto, can we uncomment these expectations?

**Review by sandbergja — CHANGES_REQUESTED** [review]
Thanks, @maxkadel !  Just some questions about commented-out expectations in the tests.

File: app/views/feedback/report_harmful_language.html.erb (hunk 1 above)
Comment (maxkadel) [reply app/views/feedback/report_harmful_language.html.erb:1]: Thanks!

File: spec/forms/report_harmful_language_form_spec.rb (hunk 2 above)
Comment (maxkadel) [reply spec/forms/report_harmful_language_form_spec.rb]: Done! Good catch!

File: spec/mailers/contact_mailer_spec.rb (hunk 3 above)
Comment (maxkadel) [reply spec/mailers/contact_mailer_spec.rb]: Ditto! Done!

**Review by sandbergja — APPROVED** [review]
Thanks, @maxkadel !

**Pull request merged**